relayer/app/sender.go

```diff
@@ -127,67 +138,75 @@ func (s Sender) SendTransaction(ctx context.Context, sub xchain.Submission) erro
 
 	txData, err := xchain.EncodeXSubmit(xchain.SubmissionToBinding(sub))
 	if err != nil {
-		return err
+		return returnErr(err)
 	}
 
 	// Reserve a nonce here to ensure correctly ordered submissions.
 	nonce, err := s.txMgr.ReserveNextNonce(ctx)
 	if err != nil {
-		return err
+		return returnErr(err)
 	}
 
 	estimatedGas := s.gasEstimator(s.chain.ID, sub.Msgs)
 
 	candidate := txmgr.TxCandidate{
 		TxData:   txData,
-		To:       &s.portal,
+		To:       &s.chain.PortalAddress,
 		GasLimit: estimatedGas,
 		Value:    big.NewInt(0),
 		Nonce:    &nonce,
 	}
 
-	tx, rec, err := s.txMgr.Send(ctx, candidate)
-	if err != nil {
-		return errors.Wrap(err, "failed to send tx", reqAttrs...)
-	}
+	asyncResp := make(chan error, 1) // Actual async response populated by goroutine below.
+	go func() {
+		tx, rec, err := s.txMgr.Send(ctx, candidate)
```

Comment:
we should close asyncResp channel, to prevent routine leak
```suggestion
		defer close(asyncResp)
```

Reply:
no goroutines to leak